test/unit/test_idstorage.c

```diff
@@ -0,0 +1,71 @@
+#include <r_util.h>
+#include "minunit.h"
+
+bool test_r_id_storage_add(void) {
+	char *str = "lol";
+	RIDStorage *ids = r_id_storage_new (5, 23);
+	ut32 id;
+	r_id_storage_add (ids, str, &id);
+	void *ptr = r_id_storage_get (ids, id);
+	r_id_storage_free (ids);
+	mu_assert_ptreq (ptr, str, "id_storage_add");
+	mu_end;
+}
+
+bool test_r_id_storage_set(void) {
+	char *str = "lol";
+	RIDStorage *ids = r_id_storage_new (5, 23);
+	r_id_storage_set (ids, str, 1);
+	void *ptr = r_id_storage_get (ids, 1);
+	r_id_storage_free (ids);
+	mu_assert_ptreq (ptr, str, "id_storage_set");
+	mu_end;
+}
+
+bool test_r_id_storage_delete(void) {
+	RIDStorage *ids = r_id_storage_new (5, 23);
+	ut32 id;
+	r_id_storage_add (ids, "lol", &id);
+	r_id_storage_delete (ids, id);
```

Comment:
Check return value deleting a valid and an invalid fd

Reply:
this would be 2 more tests, ugh

Reply:
delete is void. why would you care if delete returns success indicator? This test should test if an entry exists after deletion.

Reply:
To ensure the element u tried to delete was there already. Checking and deleting can be racy

Reply:
this whole api is racy, no mutexes anywhere. we can make this threadsafe at some point, but that would be out of scope of this pr.
What I want to say is, I'm fine with changing delete. but I don't want to do this now. Feel free to modify the pr, if you really think this should be part of this pr